pkg/maps/ipcache/ipcache.go

```diff
@@ -210,13 +230,16 @@ func (m *Map) supportsDelete() bool {
 		m.deleteSupport, _ = m.delete(invalidEntry, false)
 		log.Debugf("Detected IPCache delete operation support: %t", m.deleteSupport)
 
+		// Detect dump support
+		err := m.Dump(map[string][]string{})
+		m.dumpSupport = err == nil
+		log.Debugf("Detected IPCache dump operation support: %t", m.dumpSupport)
+
 		// In addition to delete support, ability to dump the map is
 		// also required in order to run the garbage collector which
 		// will iterate over the map and delete entries.
 		if m.deleteSupport {
-			err := m.Dump(map[string][]string{})
-			m.deleteSupport = err == nil
-			log.Debugf("Detected IPCache dump operation support: %t", m.deleteSupport)
+			m.deleteSupport = m.dumpSupport
 		}
 
 		if !m.deleteSupport {
```

Comment:
if we run SupportsDelete and SupportsDump, won't the message on the line below be printed twice?

Reply:
No, as the above code is executed via `sync.Once()`.